File: AEDS II/codigos/2 - palindromo/palindromo.c
#include <stdio.h>
#include <string.h>

void palindromo(char palavra[]){
  int ePalindromo = 1;
  int tamanho = strlen(palavra);
  for(int i=0, j = tamanho-1; i<j; i++, j--){
   if(palavra[i] != palavra[j]){
     ePalindromo = 0;
     i = j;
   }
  }
  if(ePalindromo){ printf("SIM\n"); }
  else{ printf("NAO\n"); }
  //printf("%s\n", palavra);
}

int main(){
  char palavra[500]; // 499 tem que ter um \n no final
  scanf(" %[^\n]", palavra);

  while(strcmp(palavra, "FIM")){
    palindromo(palavra);
    scanf(" %[^\n]", palavra);
  }
  return 0;
}

File: AEDS II/trabalhos/TP01Q01 - Palindromo/Palindromo.java
import java.util.Scanner;

class Palindromo{
  public static void palindromo(String palavra){
    int tamanho = palavra.length();
    boolean ePalindromo = true;

    for(int i=0; i < tamanho/2; i++){
      if(palavra.charAt(i) != palavra.charAt(tamanho-1-i)){
        ePalindromo = false;
        i = tamanho;
      }
    }

    if(ePalindromo){ System.out.println("SIM"); }
    else{ System.out.println("NAO"); }
   // System.out.println(palavra);
  }
  public static void main(String[] args){
    Scanner teclado = new Scanner(System.in);
      String palavra = teclado.nextLine();
      while(!palavra.equals("FIM")){
        palindromo(palavra);
        palavra = teclado.nextLine();
      }
    teclado.close();
  }
}
